test/download-file.spec.js

```diff
@@ -0,0 +1,56 @@
+import expect from 'expect';
+import sinon from 'sinon';
+import fs from 'fs';
+
+import { writeImageOnDisk } from '../src/download-file';
+
+/* eslint-disable  func-names, prefer-arrow-callback */
+describe('writeImageOnDisk', function() {
+  beforeEach(function() {
+    this.sandbox = sinon.sandbox.create();
+  });
+
+  afterEach(function() {
+    this.sandbox.restore();
+  });
+
+  it('should set encoding', function() {
+    const { sandbox } = this;
+    const response = {
+      setEncoding: sandbox.stub(),
```

Comment:
If we 'overwrite' the response `setEncoding` method with `sandbox.stub()` we will be able to check if it was called or not, also we can check how many times that method was called and know the arguments for each call.
That being said we will be 100% sure about what we pass and how we pass it.